web/src/custom_profile_fields_ui.js

```diff
@@ -0,0 +1,176 @@
+import $ from "jquery";
+
+import render_settings_custom_user_profile_field from "../templates/settings/custom_user_profile_field.hbs";
+
+import {$t} from "./i18n";
+import {page_params} from "./page_params";
+import * as people from "./people";
+import * as pill_typeahead from "./pill_typeahead";
+import * as typeahead_helper from "./typeahead_helper";
+import * as user_pill from "./user_pill";
+
+export function append_custom_profile_fields(element_id, user_id) {
+    const person = people.get_by_user_id(user_id);
+    if (person.is_bot) {
+        return;
+    }
+    const all_custom_fields = page_params.custom_profile_fields;
+    const all_field_types = page_params.custom_profile_field_types;
+
+    const all_field_template_types = new Map([
+        [all_field_types.LONG_TEXT.id, "text"],
+        [all_field_types.SHORT_TEXT.id, "text"],
+        [all_field_types.SELECT.id, "select"],
+        [all_field_types.USER.id, "user"],
+        [all_field_types.DATE.id, "date"],
+        [all_field_types.EXTERNAL_ACCOUNT.id, "text"],
+        [all_field_types.URL.id, "url"],
+        [all_field_types.PRONOUNS.id, "text"],
+    ]);
+
+    for (const field of all_custom_fields) {
+        let field_value = people.get_custom_profile_data(user_id, field.id);
+        const is_select_field = field.type === all_field_types.SELECT.id;
+        const field_choices = [];
+
+        if (field_value === undefined || field_value === null) {
+            field_value = {value: "", rendered_value: ""};
+        }
+        if (is_select_field) {
+            const field_choice_dict = JSON.parse(field.field_data);
+            for (const choice in field_choice_dict) {
+                if (choice) {
+                    field_choices[field_choice_dict[choice].order] = {
+                        value: choice,
+                        text: field_choice_dict[choice].text,
+                        selected: choice === field_value.value,
+                    };
+                }
+            }
+        }
+
+        const html = render_settings_custom_user_profile_field({
+            field,
+            field_type: all_field_template_types.get(field.type),
+            field_value,
+            is_long_text_field: field.type === all_field_types.LONG_TEXT.id,
+            is_user_field: field.type === all_field_types.USER.id,
+            is_date_field: field.type === all_field_types.DATE.id,
+            is_url_field: field.type === all_field_types.URL.id,
+            is_pronouns_field: field.type === all_field_types.PRONOUNS.id,
+            is_select_field,
+            field_choices,
+            for_manage_user_modal: element_id === "#edit-user-form .custom-profile-field-form",
+        });
+        $(element_id).append(html);
+    }
+}
+
+export function initialize_custom_user_type_fields(
+    element_id,
+    user_id,
+    is_editable,
+    pill_update_handler,
```

Comment:
We passed a bool value here previously, but changed to directly pass the function to call on updating pill input to avoid moving update functions like `update_user_custom_profile_fields` here as they are only used for "Profile" settings panel and not for the profile fields shown in user edit form.

Reply:
Yeah a callback makes sense here.